lib/OpenQA/Schema/Result/Jobs.pm

```diff
@@ -385,12 +385,13 @@ sub prepare_for_work ($self, $worker = undef, $worker_properties = {}) {
         $job_hashref->{settings}->{NICVLAN} = join(',', @vlans);
     }
 
-    # assign new tmpdir, clean previous one
-    if (my $tmpdir = $worker->get_property('WORKER_TMPDIR')) {
-        File::Path::rmtree($tmpdir);
+    unless ($worker_properties->{WORKER_TMPDIR}) {
+        # assign new tmpdir, clean previous one
+        if (my $tmpdir = $worker->get_property('WORKER_TMPDIR')) {
+            File::Path::rmtree($tmpdir);
+        }
+        $worker->set_property(WORKER_TMPDIR => tempdir());
```

Comment:
This might be good to move into a little helper function, since you're already doing this in two places

Reply:
I checked them. The code are different staring from line 393 in two places. It does not look right to just leave a conditional  statement in a function.